homeassistant/components/compit/__init__.py

```diff
@@ -0,0 +1,60 @@
+"""The Compit integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+
+from compit_inext_api import CompitAPI, DeviceDefinitionsLoader
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, PLATFORMS
+from .coordinator import CompitDataUpdateCoordinator
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Compit from a config entry."""
+
+    session = async_get_clientsession(hass)
+    api = CompitAPI(entry.data["email"], entry.data["password"], session)
+    gates = await api.authenticate()
```

Comment:
```suggestion
    system_info = await api.authenticate()
```
looking at the `authenticate` method, this returns a SystemInfo, which contains a `gate` property, so isn't `system_info` a better name?

Reply:
on config_flow we have `try/catch` for this call, should we add it here too?